EIPS/eip-7015.md

```diff
@@ -0,0 +1,184 @@
+
+Creator consent is given by signing an [EIP-712](./eip-712.md) compatible message; all signatures compliant with this EIP MUST include all fields defined. The struct signed is:
+
+```solidity
+struct CreatorAttribution {
+	string name;
+	string symbol;
+	bytes32 salt;
+	address token;
+}
+```
+
+Where `name` and `symbol` are the ERC-721 metadata parameters, `salt` is an additional randomness parameter, and `token` is the NFTs address that will be deployed. Any extra parameters used to deploy the NFT contract MAY be hashed into the `salt` parameter.
+
+**Signature Validation**
+
+Creator attribution is given through a signature verification that MUST be verified by the NFT contract and an event that MUST be emitted by the NFT contract during the deployment transaction. The event includes all the necessary fields for reconstructing the signed digest and validating the signature to ensure it matches the specified creator. The event name is `CreatorAttribution` and includes the following fields:
+
+- `name`: ERC-721Metadata name parameter
+- `symbol`: ERC-721Metadata symbol parameter
+- `salt`: an extra randomness parameter that encodes other relevant information for deploying the NFT contract
+- `domainName`: the domain name of the contract verifying the singature (for EIP-712 signature validation)
+- `version`: the version of the contract verifying the signature (for EIP-712 signature validation)
+- `creator`: the account address for the creator of the NFT
+- `signature`: the creator’s signature
+
+The event is defined as follows:
+
+```solidity
+event CreatorAttribution(
+	string name,               // <-------------------------------------
+	string symbol,             //      Parameters signed by creator     |
+	bytes32 salt,              //                                      |
+	string domainName,         // <--- EIP-712 Domain Data -------------
+	string version,            // <-------------------------------------
+	address creator,           // <--- Creator --------------------------
+	bytes signature            // <--- Creator Signature ----------------
+);
+```
+
+Note that although the `chainId` parameters is necessary for [EIP-712](./eip-712.md) signatures, we omit the parameter from the event as it can be inferred through the transaction data. Similarly, the `token` parameter in the `CreatorAttribution` stuct MUST match the address of the `emitter` field of the event, hence we omit it as well.
+
+A platform can verify the validity of the creator attribution by reconstructing the signature digest with the parameters emitted and recovering the signer from the `signature` parameter. The `creator` parameter MUST match the recovered signer from the `signature` parameter. After verification, platforms can safely attribute the `creator` address as the creator instead of the account that submitted the transaction.
+
+### Reference Implementation
+
+#### Example signature validator
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code inline in the EIP must be licensed under `CC0-1.0`. Code in the assets directory can be `MIT` if you'd like.

Reply:
Fixed!